### PR TITLE
Weapon events refactor.

### PR DESCRIPTION
**Pull request by gir489returns**
Removed legacy REMOVE_ALL_WEAPONS_EVENT network event check.
Added GIVE_WEAPON_EVENT check to prevent people giving us weapons.

### CODE REVIEW

**Comment by lonelybud** [comment]
we dont generally receive weapons from the player during activity, job, heist etc right?

File: src/hooks/protections/received_event.cpp [hunk 1]
```diff
@@ -582,23 +582,26 @@ namespace big
 
 			if (g_local_player && g_local_player->m_net_object && g_local_player->m_net_object->m_object_id == net_id)
 			{
+				weapon_item weapon = g_gta_data_service->weapon_by_hash(hash);
 				g_notification_service->push_warning("PROTECTIONS"_T.data(),
-				    std::vformat("REMOVE_WEAPON_ATTEMPT"_T, std::make_format_args(source_player->get_name())));
+					std::format("{} {} {}.", source_player->get_name(), "REMOVE_WEAPON_ATTEMPT"_T, weapon.m_display_name));
```
Comment (Yimura) [inline src/hooks/protections/received_event.cpp]: This change will fuck over every other language that had this key translated.

File: src/hooks/protections/received_event.cpp [hunk 2]
```diff
@@ -582,23 +582,26 @@ namespace big
...
-			int net_id = buffer->Read<int>(13);
+			std::int16_t net_id = buffer->Read<std::int16_t>(13);
+			Hash hash           = buffer->Read<Hash>(32);
 
 			if (g_local_player && g_local_player->m_net_object && g_local_player->m_net_object->m_object_id == net_id)
 			{
+				weapon_item weapon = g_gta_data_service->weapon_by_hash(hash);
 				g_notification_service->push_warning("PROTECTIONS"_T.data(),
-				    std::vformat("REMOVE_ALL_WEAPONS_ATTEMPT"_T, std::make_format_args(source_player->get_name())));
+				    std::format("{} {} {}.", source_player->get_name(), "GIVE_WEAPON_ATTEMPT"_T, weapon.m_display_name));
```
Comment (Yimura) [inline src/hooks/protections/received_event.cpp]: This change will fuck over every other language that had this key translated.

**Review by Yimura — CHANGES_REQUESTED** [review]

File: src/hooks/protections/received_event.cpp (hunk 1 above)
Comment (gir489returns) [reply src/hooks/protections/received_event.cpp:587]: What should the remedy be? Change the key?

File: src/hooks/protections/received_event.cpp (hunk 2 above)
Comment (gir489returns) [reply src/hooks/protections/received_event.cpp:604]: No it won't, this key is new.

File: src/hooks/protections/received_event.cpp (hunk 1 above)
Comment (Yimura) [reply src/hooks/protections/received_event.cpp]: Yeah as it will just be incorrect in other languages since the format doesn't work.

**Review by Yimura — APPROVED** [review]

**Pull request merged**